lib/functions/transfer.js

```diff
@@ -82,9 +82,31 @@ class Transfer extends AsyncGeneratorFunction {
                         headers: Object.assign({
                             [HTTP.HEADER.CONTENT_LENGTH]: blob.size,
                             [HTTP.HEADER.CONTENT_TYPE]: transferPart.metadata.contentType
-                        }, transferPart.headers)
+                        }, transferPart.targetHeaders)
                     });
                 }, this.options);
+            } else if (targetUrl && isFileProtocol(targetUrl) && transferPart.source.url) {
+                await retry(async () => {
+                    const totalSize = transferPart.metadata.contentLength;
+                    const response = await streamGet(transferPart.source.url, {
```

Comment:
When the code here executes, was the URL validity already verified?

Reply:
To fail early, and with a usable error.

Reply:
Yes, it's supposed to be checked when you create `Asset` and then when the `TransferPart` are created.